src/Stratis.Bitcoin.Features.Consensus/CoinViews/ICachedCoinView.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using NBitcoin;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.CoinViews
+{
+    /// <summary>
+    /// Database of UTXOs.
+    /// </summary>
+    public interface ICachedCoinView : ICoinView
```

Comment:
With regards to naming, what about `ICoinViewRewindData` ?

Reply:
Problem is it is not just a rewinddata.